core/src/main/java/fr/sncf/osrd/speedcontroller/generators/ConstructionAllowanceGenerator.java

```diff
@@ -48,15 +46,42 @@ protected double getFirstGuess() {
                                                        double targetSpeed,
                                                        double initialPosition,
                                                        double endPosition) throws SimulationError {
+
+        // for later implementation, speed limit under which a train would consume too much capacity
+        double capacitySpeedLimit = 30 / 3.6;
+
         var currentSpeedControllers = new HashSet<>(maxSpeedControllers);
 
         // running calculation to get the initial speed and final speed of the Region Of Interest (ROI)
         var expectedSpeeds = getExpectedSpeeds(schedule, currentSpeedControllers, TIME_STEP,
                 0, endPosition, schedule.initialSpeed);
         var initialSpeed = expectedSpeeds.interpolate(initialPosition);
+        var endSpeed = expectedSpeeds.interpolate(endPosition);
+        double accelerationFirstPosition = endPosition;
         var res = new HashSet<>(maxSpeedControllers);
 
-        // coasting phase
+        if (targetSpeed < endSpeed) {
+            // if the target speed is under the final speed of the Region of Interest (RoI)
+            // that means there will be an acceleration phase at the end of the RoI to catch up with endSpeed
+            var acceleratingPhase =
+                    generateAcceleratingCurveBackwards(schedule, initialPosition, endSpeed, endPosition, targetSpeed);
+            // memorize the position where the accelerating curve crosses targetSpeed, if there is one
+            for (var element : acceleratingPhase.entrySet()) {
+                if (element.getValue() <= targetSpeed && accelerationFirstPosition == endPosition)
+                    accelerationFirstPosition = element.getKey();
+            }
```

Comment:
Instead of the condition `accelerationFirstPosition == endPosition`you can `break`once you found the intersection.

Reply:
Ah, right, i forgot i could do that ! Thank you !